README.md

```diff
@@ -1,86 +1,149 @@
+## Building from source
+
+Building MIGraphX from sources needs dependencies, which must be installed before building the source code. In the
+following, we first list the dependencies, and then explain two ways of installing them.
+
+### List of dependencies
+The following is a list of dependencies required to build MIGraphX from source code. This list is also available in the
+requirement files ```dev-requirements.txt``` and ```requirements.txt```.
 
-## Prerequisites
 * [ROCm cmake modules](https://github.com/RadeonOpenCompute/rocm-cmake) **required**
 * [MIOpen](https://github.com/ROCmSoftwarePlatform/MIOpen) for running on the GPU
+* [rocBLAS](https://github.com/ROCmSoftwarePlatform/rocBLAS) for running on the GPU
 * [HIP](https://github.com/ROCm-Developer-Tools/HIP) for running on the GPU
 * [Protobuf](https://github.com/google/protobuf) for reading [onnx](https://github.com/onnx/onnx) files
 * [Half](http://half.sourceforge.net/) - IEEE 754-based half-precision floating point library
 * [pybind11](https://pybind11.readthedocs.io/en/stable/) - for python bindings
+* [ONNX 1.7.0](https://github.com/onnx/onnx) and [Pytest](https://github.com/pytest-dev/pytest) for running the ONN backend
+  tests 
 
-## Installing the dependencies
+Note: 1) We have to use ONNX version 1.7.0 since changes in ONNX 1.8.0 is incompatible with our current implementation. 
+ONNX 1.7.0 can be installed as ```pip3 install onnx==1.7.0```, and updates to support ONNX version 1.8.0 will come soon. 2)
+MIOpen and rocBLAS can be installed as ```sudo apt update && sudo apt install -y miopen-hip rocblas```.
 
-Dependencies can be installed using the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
+### Installing the dependencies
+There are two alternative ways to install the above dependencies:
 
-To install rbuild:
-```
-pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
-```
+#### Run a shell script [build_prereqs.sh](./tools/build_prereqs.sh).
+
+(Note: 1. You need the sudo to install the above dependencies. 2. All dependencies are installed in default locations in
+the system and are accessible by all users.)
+
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-To build dependencies along with MIGraphX
+To install rbuild (sudo may be needed.):
 ```
-rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`
 
-## Building MIGraphX from source
+Dependencies are listed in files `requirements.txt` and `dev-requirements.txt` in the project diretory. To build the
+dependencies,
+* In ROCm3.3:
 
-## Configuring with cmake
-
-First create a build directory:
+```
+rbuild develop -d depend -B build --cxx=/opt/rocm/bin/hcc
```

Comment:
Hcc is deprecated so I would just remove this from the readme.

Reply:
If someone use a old GPU like the gfx803, rocm3.3 maybe needed as we mentioned in yesterday's team meeting.

Reply:
opinions from other guys are appreciated.

Reply:
> If someone use a old GPU like the gfx803, rocm3.3 maybe needed as we mentioned in yesterday's team meeting.

I think we should just have wiki page to go over using migraphx with older rocm 3.3 then. We only test migraphx with hcc to use ubsan, however once clang supports ubsan then we will drop testing and support for rocm 3.3.

Reply:
added a wiki page for rocm3.3 at: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/wiki/Build-MIGraphX-Source-in-ROCm3.3